extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+from typing import Any, Dict, Literal, Optional
+
+import firebase_admin
+from firebase_admin import credentials, firestore_async
+from google.cloud import firestore
+from google.cloud.firestore_v1.base_query import FieldFilter
+from pydantic import BaseModel
+
+import models
+
+from .. import datastore
+
+
+class Config(BaseModel, datastore.AbstractConfig):
+    kind: Literal["firestore"]
+    projectId: str
+    serviceAccountId: str
+
```

Comment:
1. I think we can let projectId be optional -- if it's unset we shouldn't pass into the client and just let it use the default
2. I don't think we need service account id

Reply:
Made projectId optional. Deleted service account id field.